eos-updater/poll-common.c

```diff
@@ -513,11 +514,15 @@ should_follow_checkpoint (OstreeSysroot     *sysroot,
 {
   g_autoptr(GHashTable) hw_descriptors = NULL;
   const gchar *sys_vendor, *product_name;
-  /* https://phabricator.endlessm.com/T32542, https://phabricator.endlessm.com/T32552 */
+  /* https://phabricator.endlessm.com/T32542,
+   * https://phabricator.endlessm.com/T32552,
+   * https://phabricator.endlessm.com/T33311 */
   gboolean is_conditional_upgrade_path =
     (g_str_equal (booted_ref, "eos3a") ||
      g_str_has_suffix (booted_ref, "/eos3a") ||
-     g_str_has_suffix (booted_ref, "nexthw/eos3.9"));
+     g_str_has_suffix (booted_ref, "nexthw/eos3.9") ||
+     g_str_has_suffix (booted_ref, "latest1") ||
+     g_str_has_suffix (booted_ref, "eos4"));
```

Comment:
Likewise, I think this should be `/eos4`. However, I wonder if we should add it at all. It would make the 2nd commit unneeded since the `eos4` branch wouldn't be considered for upgrade at all. On the other hand, if we later do add an `eos4` to `eos4a` upgrade, we would want the `eos4` to `latest1` block you've added here.

Reply:
If we are not going to have `eos4a` and `eos4` branch will not upgrade right now, then we can drop the 2nd commit.